### PR TITLE
ci: add GH actions setup

### PR DESCRIPTION
**Pull request by Killusions**
Adds the setup for GitHub actions and releasing. To release start the action manually. Tested here: https://github.com/Killusions/lint

### CODE REVIEW

**Comment by Killusions** [comment]
This needs 2 secrets:

1. NPM_TOKEN
2. REPO_TOKEN (a token from GitHub with write perms on the repo (default CI token doesn't))

**Comment by Killusions** [comment]
Just tested the [test package published on my namespace](https://www.npmjs.com/package/@killusions/eslint-config-angular) and it works.

**Comment by kfenner** [comment]
@Killusions Thank you very much 🙏 

@fh1ch Maybe you also want to take a look here. You'd also need to provide the secrets I guess 🙏 🚀

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  workflow_dispatch:
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
```
Comment (spike-rabbit) [inline .github/workflows/release.yml:13]: I understand fetch-depth here, buy why persist-credentials: false?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [22.x]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (Killusions) [reply .github/workflows/build.yml:18]: Yes for commitlint, I added it later

**Review by timowolf — CHANGES_REQUESTED** [review]
We cannot really review and merge unless we

- align on the release and maintenance process with the stakeholders
- document the release process in a markdown file in the repo

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [22.x]
```
[the remark this answers is not in the document]
Comment (Killusions) [reply .github/workflows/build.yml]: Removed

File: .github/workflows/release.yml (hunk 1 above)
Comment (Killusions) [reply .github/workflows/release.yml]: Removed

**Comment by Killusions** [comment]
Switched to the default semantic release workflow.

**Comment by timowolf** [comment]
@Killusions @kfenner @spike-rabbit (cc @fh1ch ) Here is what we should do in the PR

- [ ] We follow the semantic release as close as possible
- [ ] Do not push a commit to main on a release
- [ ] Release on every merge on main
- [ ] Keep version 0.0.0-development in the package.json in the code
- [ ] Replace the 0.0.0-development version number in the NPM releases and also in the archives we can download from GitHub with the real version. The downloaded zip files should contain the correct version number (Something that seems wrong here https://github.com/semantic-release/semantic-release/releases/tag/v24.2.0).  
- [ ] Create a configuration for release candidates on a branch called `next`. Please check if we need to create a next branch for every major release or if we can rebase it on main.
- [ ] Add the NPM and GitHub token to the repo (I can do this )
- [ ] Merge this PR into `next` branch to test if this setup works, by creating 0.1..0-rc{0} release candidates and be able to test this PR end to end. Thus we should verify the npm deployment as well as the release numbers and the commit messages in the changelog
- [ ] Document our process in a markdown file
- [ ] When all good, we merge next in main

Otherwise we cannot test if the PR works. What do you think?

File: CHANGELOG.md
Comment (timowolf) [inline CHANGELOG.md:1]: I am always looking for a changelog file. Maybe we keep the file and point to the GitHub releases. But we can also add this later.

Comment (kfenner) [reply CHANGELOG.md:1]: @Killusions For me your changes look good 👍 

@timowolf Agree with everything, maybe the documentation we can do in a follow up when we know all has worked like it should?

**Comment by timowolf** [comment]
@Killusions Many thanks for the great work. We go ahead with #2  and test it on a next branch first. I am closing this one.

**Pull request closed without merging**